homeassistant/components/control4/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow for Control4 integration."""
+from asyncio import TimeoutError as asyncioTimeoutError
+import logging
+
+from aiohttp.client_exceptions import ClientError
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import NotFound, Unauthorized
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_CONTROLLER_UNIQUE_ID, DEFAULT_SCAN_INTERVAL, MIN_SCAN_INTERVAL
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class Control4Validator:
+    """Validates that config details can be used to authenticate and communicate with Control4."""
+
+    def __init__(self, host, username, password):
+        """Initialize."""
+        self.host = host
+        self.username = username
+        self.password = password
+        self.controller_unique_id = None
+        self.director_bearer_token = None
+
+    async def authenticate(self) -> bool:
+        """Test if we can authenticate with the Control4 account API."""
+        try:
+            account = C4Account(self.username, self.password)
+            # Authenticate with Control4 account
+            await account.getAccountBearerToken()
+
+            # Get controller name
+            account_controllers = await account.getAccountControllers()
+            self.controller_unique_id = account_controllers["controllerCommonName"]
+
+            # Get bearer token to communicate with controller locally
+            self.director_bearer_token = (
+                await account.getDirectorBearerToken(self.controller_unique_id)
+            )["token"]
+            return True
+        except (Unauthorized, NotFound):
+            return False
```

Comment:
It would be great if we could distinguish for the user in the error message if he has provided the incorrect credentials or if something could not be found.

Reply:
I think I'll just stop catching `NotFound` - I can't imagine a scenario where we just retrieved the controller name from Control4, then immediately after it can't be found.

Reply:
Just let the user know and they can handle it accordingly.